kernel/Utilities/GlobalIndexing.cpp

```diff
@@ -58,6 +58,64 @@ GlobalIndexing::GlobalIndexing(Base::MeshManipulatorBase *mesh, Layout layout,
     reset(mesh, layout, unknowns);
 }
 
+/**
+ * Visit the element and all its boundary parts in the canonical order
+ *
+ * @param element The element to visit
+ * @param meshDimension The dimension of the mesh
+ * @param visitor The visitor to use
+ */
+void visitElementParts(const Base::Element *element, std::size_t meshDimension,
+                       std::function<void(const Base::MeshEntity *)> visitor) {
```

Comment:
you could also just make this a templated function and get rid of the std::function.

Reply:
if this is performance critical, you put in lambdas mostly anyway.

Reply:
Yes that is a good idea.